packages/aws-cdk-lib/aws-codepipeline/README.md

```diff
@@ -419,6 +421,52 @@ for details on how to use the variables for each action class.
 See the [CodePipeline documentation](https://docs.aws.amazon.com/codepipeline/latest/userguide/reference-variables.html)
 for more details on how to use the variables feature.
 
+### Pipeline-level variables
+
+You can add one or more variables at the pipeline level. You can reference
+this value in the configuration of CodePipeline actions. You can add the
+variable names, default values, and descriptions when you create the pipeline.
+Variables are resolved at the time of execution.
+
+Note that using pipeline-level variables in any kind of Source action is not supported.
```

Comment:
I'm confused about this sentence. it makes sense, because we're talking about pipeline-level variables and not action-level variables. Just via the name, I'd assume that pipeline-level variables are unusable in any kind of action (not just source actions). Is that a correct assumption?

Reply:
My confusion is that pipeline-level variables cannot be USED in source action but can be USED elsewhere. Of course. the scope of pipeline-level is the level its defined in, not where its used in. My bad